certs/certs.go

```diff
@@ -99,10 +109,33 @@ func ValidateRoot(certStore trustmanager.X509Store, root *data.Signed, gun strin
 			return &ErrValidationFail{Reason: "failed to validate data with current trusted certificates"}
 		}
 	} else {
-		logrus.Debugf("found no currently valid root certificates for %s", gun)
+		logrus.Debugf("found no currently valid root certificates for %s, using trust_pinning config to bootstrap trust:", gun, trustPinning)
+		trustPinChecker, err := NewTrustPinChecker(trustPinning, gun)
+		if err != nil {
+			return &ErrValidationFail{Reason: err.Error()}
+		}
+
+		validPinnedCerts := []*x509.Certificate{}
+		for _, cert := range allValidCerts {
+			_, intermediateCerts := parseAllCerts(signedRoot)
```

Comment:
Yep definitely, I'll refactor `validateRootLeafCerts` to not need to call `parseAllCerts` - that's a great idea